tower_cli/cli/transfer/send.py

```diff
@@ -1144,6 +1139,19 @@ def import_roles(self, existing_asset, new_roles, asset_type):
                             actor, item_to_add, role['name'], e)
                         )
 
+    def get_role_type(self, type, name):
+        role_type = type
+        if role_type == 'Ad Hoc':
+            role_type = 'adhoc'
```

Comment:
Firstly, this here looks like an AWX bug. I see that you get `type` from the serialization (unless you modified it somewhere) which is

```json
            "type": "role",
```

and if you saw "Ad Hoc" there, that needs to be filed upstream to AWX. That's quite concerning.

Reply:
The method is trying to determine the internal role name. It first assumes that you correctly passed it the internal role type (i.e. adhoc). But if you gave it "role" as the type and then the name 'Ad Hoc' it has to convert it to adhoc.

Reply:
I think the API really should be providing the internal role name, but since it doesn't I'm starting to follow what you did here a little better.